src/machine/asm/mod.rs

```diff
@@ -433,9 +432,9 @@ impl AotCode {
     }
 }
 
-pub struct AsmMachine<'a> {
-    pub machine: DefaultMachine<'a, Box<AsmCoreMachine>>,
-    pub aot_code: Option<&'a AotCode>,
+pub struct AsmMachine {
+    pub machine: DefaultMachine<Box<AsmCoreMachine>>,
+    pub aot_code: Option<Rc<AotCode>>,
```

Comment:
I think it's better to use `Arc` here. It's not that ckb-vm will be using threads, it's more that if you think about the usage patterns, it's likely the same binary being AOTed could be shared amongst multiple ckb-vm instances, all using the same `AotCode` instance to speed up execution. In this case, `AotCode` might be shared among multiple threads. Using `Rc` here severely limits the ability to share compiled immutable code.

Reply:
Right.